pkg/apis/pipeline/v1beta1/taskrun_conversion.go

```diff
@@ -193,6 +199,150 @@ func (trso *TaskRunSidecarOverride) convertFrom(ctx context.Context, source v1.T
 	trso.Resources = source.ComputeResources
 }
 
+// ConvertTo implements apis.Convertible
+func (trs *TaskRunStatus) ConvertTo(ctx context.Context, sink *v1.TaskRunStatus) error {
+	// TODO: handle duckv1beta1 to v1
```

Comment:
The conversion from [duckv1beta1.Status](https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1beta1/taskrun_types.go#L136) to [duckv1.Status] is not provided. The [ConvertTo](https://github.com/tektoncd/pipeline/blob/main/vendor/knative.dev/pkg/apis/duck/v1beta1/status_types.go#L109) made me wonder if we should have duckv1.

Reply:
Hum, I am not sure I understand the `ConvertTo` function on `v1beta1.Status` 😅.
I think for our `v1` API, using `v1.Status` is fine, and maybe handling the conversion ourselves works ? 🤔 

cc @imjasonh @mattmoor

Reply:
The current conversion is implemented on a non-pointer basis. Put as a TODO in the comment and this shall not be a blocker for the v1 release process or this PR.